spec/researches/buildKeywordFormsSpec.js

```diff
@@ -1,89 +1,204 @@
-		expect( result ).toContain( "one" );
-		expect( result ).toContain( "ones" );
-		expect( result ).toContain( "and" );
-		expect( result ).toContain( "ands" );
-		expect( result ).toContain( "two" );
-		expect( result ).toContain( "twos" );
+describe( "A test for building keyword and synonyms forms for a paper", function() {
+	it( "returns the exact matches if the input strings are embedded in quotation marks and word forms if not; for empty locale", function() {
+		const keyword = "\"I am going for a walk\"";
+		const synonyms = "\"You are not going for a walk\", You are going for a movie, And he is going to work diligently.";
+		let locale;
+
+		const expectedResult = {
+			keyphraseForms: [ [ "I am going for a walk" ] ],
+			synonymsForms: [
+				[ [ "You are not going for a walk" ] ],
+				[ movieForms ],
+				[ workForms, diligentlyForms ],
+			],
+		};
+		expect( collectForms( keyword, synonyms, locale, true ) ).toEqual( expectedResult );
+	} );
+
+	it( "returns the exact matches if the input strings are embedded in quotation marks and word forms if not; for English", function() {
+		const keyword = "\"I am going for a walk\"";
+		const synonyms = "\"You are not going for a walk\", You are going for a movie, And he is going to work diligently.";
+		const locale = "en_EN";
+
+		const expectedResult = {
+			keyphraseForms: [ [ "I am going for a walk" ] ],
+			synonymsForms: [
+				[ [ "You are not going for a walk" ] ],
+				[ movieForms ],
+				[ workForms, diligentlyForms ],
+			],
+		};
+		expect( collectForms( keyword, synonyms, locale, true ) ).toEqual( expectedResult );
+	} );
+
+	it( "returns the exact matches if the input strings are embedded in quotation marks and single-word arrays if not; for French (no morphology yet)", function() {
+		const keyword = "Je vais me promener";
+		const synonyms = "\"Tu ne vas pas te promener\", Tu vas voir un film, Et lui il va travailler dur.";
+		const locale = "fr_FR";
+
+		const expectedResult = {
+			keyphraseForms: [ [ "promener" ] ],
+			synonymsForms: [
+				[ [ "Tu ne vas pas te promener" ] ],
+				[ [ "voir" ], [ "film" ] ],
+				[ [ "travailler" ], [ "dur" ] ],
+			],
+		};
+		expect( collectForms( keyword, synonyms, locale, true ) ).toEqual( expectedResult );
+	} );
+
+	it( "returns the exact matches if the input strings are embedded in quotation marks and single-word arrays if not; for an unexisting locale (no morphology and function words)", function() {
+		const keyword = "\"I am going for a walk\"";
+		const synonyms = "\"You are not going for a walk\", You are going for a movie, And he is going to work diligently.";
+		const locale = "yep_YEP";
+
+		const expectedResult = {
+			keyphraseForms: [ [ "I am going for a walk" ] ],
+			synonymsForms: [
+				[ [ "You are not going for a walk" ] ],
+				[ [ "you" ], [ "are" ], [ "going" ], [ "for" ], [ "a" ], [ "movie" ] ],
+				[ [ "and" ], [ "he" ], [ "is" ], [ "going" ], [ "to" ], [ "work" ], [ "diligently" ] ],
```

Comment:
Just a note for future reference: for languages for which we don't have morphology support, we also include function words in the keyword/synonym forms (because we can't differentiate between function and content words). This means that we'll have to be careful later on to distinguish between languages with morphology support and languages without when we match keywords in texts since the things that we match are quite different.

Reply:
You probably mean "function word support". I totally agree. Maybe somewhat harsher assessment criteria for those assessments that do not require ALL words to be matched, but just some.